src/blog/exploring_aleo_record_model.md

```diff
@@ -0,0 +1,178 @@
+
+
+## Privacy
+There are generally four different types of privacy that relate to blockchains.
+
+Aleo fulfils the 3 of them:
+- [x] Private inputs (messages)
+- [x] Private outputs (state changes)
+- [x] Private user
+- [ ] Private function 
+
+Initially, Aleo was aiming for function privacy as well (as detailed in the original ZEXE paper) but decided against it as it would have led to worse performance and longer proving times. 
+
+
+
+## Comparing storage states in Blockchains
+There are two main state models used in blockchains - UTXO (unspent transaction output) and the account model (introduced by Ethereum). 
+
+Aleo uses a variation of the UTXO model - the record model.
+
+
+<p align="center" width="100%">
+<img src="./images/account_vs_utxo.png" alt="Account vs UTXO"/>
+</p>
+
+<p align="center" width="100%">
+Diagram taken from <a href="https://www.galaxy.com/insights/research/mev-how-flashboys-became-flashbots/"> here</a>
+</p>
+<br/>
+
+### Account Model
+In the account model as used in Ethereum, the application state can be found by referencing to a particular address. 
+
+As such, anyone would be able to view the activities of any account, simply with the knowledge of the address.
+
+<p align="center" width="100%">
+<img src="./images/ethereum_storage.png" alt="Ethereum Storage Diagram"/>
+</p>
+
+<p align="center" width="100%">
+Diagram taken from <a href="https://ethereum.org/en/develope.rs/docs/accounts/"> ethereum site</a>
+</p>
+<br/>
+
+
+<p align="center" width="100%">
+<img src="./images/ethereum_world_state.png" alt="Ethereum World State Diagram"/>
+</p>
+
+<p align="center" width="100%">
+Diagram taken from <a href="https://www.lucassaldanha.com/ethereum-yellow-paper-walkthrough-2/"> here</a>
+</p>
+
+<br/>
+
+
+### Record Model
+In the record model, the application state, along with it's owner are encrypted and stored on the blockchain. 
+
+
+<p align="center" width="300">
+<img src="./images/record.png" alt="Aleo Records Diagram"/>
+</p>
+
+<p align="center" width="100%">
+Diagram taken from <a href="https://eprint.iacr.org/2018/962.pdf"> ZEXE paper</a>
+</p>
+
+<br/>
+
+
+<p align="center" width="100%">
+<img src="./images/aleo_ledger.png" alt="Aleo World State Diagram"/>
+</p>
+
+
+<p align="center" width="100%">
+Diagram taken from <a href="https://eprint.iacr.org/2018/962.pdf"> ZEXE paper</a>
+</p>
+<br/>
+
+
+## Record Model Explained
+Records are a fundamental data structure that can contain any arbitrary payload and are used for encoding user assets or application states. A record represents a certain state of a program in the global state, for example, the balance of an account or your identity document. 
```

Comment:
Suggestion for a clearer way to explain records:
"...application states. Records represent a fragment of the global state kept on-chain. For example, the balance of your credits in a given account is composed by the multiple credit records that have your address as the `owner`".